langchain_core: add `file_type` option to make file type default as `png`

MasaYan24 opened the pull request:
Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [ ] **description**
langchain_core.runnables.graph_mermaid.draw_mermaid_png calls this function, but the Mermaid API returns JPEG by default. To be consistent, add the option `file_type` with the default `png` type.

- [ ] **Add tests and docs**: If you're adding a new integration, please include
With this small change, I didn't add tests and docs.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: 
One long sentence was divided into two.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

MasaYan24 committed:
add `file_type` to make file type default `png`

langchain_core.runnables.graph_mermaid.draw_mermaid_png calls this function, but the Mermaid API returns JPEG as default.
Add option `file_type` with default `png` type to be consistent.
Also background_color default was wrong and fixed with `!`

MasaYan24 committed:
Revert incorrect fix from previous commit

I mistakenly thought there was an error in the `background_color` option and added a `!` for correction. However, the `!` was already correctly added further down. This commit reverts the incorrect fix.

MasaYan24 committed:
Bug fixed: the option added as last option

The `file_type` option was added as the second option, but the other function calling the function `_render_mermaid_using_api` without option names crashes.
The `file_type` option was added as the last option, so the effect is controlled locally.

MasaYan24 committed:
Merge branch 'master' into patch-1

vbarda commented on libs/core/langchain_core/runnables/graph_mermaid.py:
```diff
@@ -306,6 +306,7 @@ def _render_mermaid_using_api(
     mermaid_syntax: str,
     output_file_path: Optional[str] = None,
     background_color: Optional[str] = "white",
+    file_type: Optional[str] = "png",
```
could you make the type `Optional[Literal["jpeg", "png", "webp"]]`

MasaYan24 replied:
Thank you for the review.
`typing.Literal` is imported and type annotation is updated as suggested.

MasaYan24 committed:
update type annotation

I have updated the type annotations to be more stringent. `Optional[str]` to `Optional[Literal["jpeg", "png", "webp"]]`.

MasaYan24 committed:
bug fixed: `Literal` is imported from typing lib.

`Literal` has not been imported. Fixed.

MasaYan24 committed:
Merge branch 'master' into patch-1
